.changeset/poor-guests-heal.md

```diff
@@ -0,0 +1,5 @@
+---
+'@wagmi/core': patch
+---
+
+Changed `waitForTransaction` behavior to throw an error if the transaction reverted.
```

Comment:
@jxom thinking this should probably be `minor`, but curious what you think about as `patch`?